test/extended/authorization/podsecurity_admission.go

```diff
@@ -189,14 +193,302 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity][Feature:SCC]", func() {
+
+				// add a role to the default SA
+				role, err := bindRoleToDefaultSA(ctx, oc, ns.Name)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// set some SCCs on the role and test PSa label outcome
+				for scc, psaLabel := range sccToPSa {
+					err = setSCCPermissions(ctx, oc, role, scc)
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					ns, err = waitAndGetNamespace(ctx, oc, ns.Name)
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					if testCase.sccChangesPSa {
+						// SCC assignment yields PSa label change by the labelsyncer
+						o.Expect(ns.GetLabels()[psapi.EnforceLevelLabel]).To(o.Equal(psaLabel))
+						o.Expect(ns.GetLabels()[psapi.AuditLevelLabel]).To(o.Equal(psaLabel))
+						o.Expect(ns.GetLabels()[psapi.WarnLevelLabel]).To(o.Equal(psaLabel))
+					} else {
+						// SCC assignment does not yield any PSa label change
+						o.Expect(ns.GetLabels()[psapi.EnforceLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterUpdate))
+						o.Expect(ns.GetLabels()[psapi.AuditLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterUpdate))
+						o.Expect(ns.GetLabels()[psapi.WarnLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterUpdate))
+					}
+				}
+			})
+		}
+	})
+})
+
+// creates a role & rolebinding for the test and binds it to the default SA
+func bindRoleToDefaultSA(ctx context.Context, oc *exutil.CLI, nsName string) (*v1.Role, error) {
+	role := &v1.Role{
+		ObjectMeta: metav1.ObjectMeta{Name: "my-role-with-scc", Namespace: nsName},
+	}
+
+	roleBinding := &v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{Name: "my-role-binding", Namespace: nsName},
+		Subjects: []v1.Subject{
+			{Kind: v1.ServiceAccountKind, Name: "default"},
+		},
+		RoleRef: v1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     role.ObjectMeta.Name,
+		},
+	}
+
+	_, err := oc.AdminKubeClient().RbacV1().Roles(nsName).Create(ctx, role, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = oc.AdminKubeClient().RbacV1().RoleBindings(nsName).Create(ctx, roleBinding, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return role, nil
+}
+
+// updates a role to use a specific SCC
+func setSCCPermissions(ctx context.Context, oc *exutil.CLI, role *v1.Role, scc string) error {
+	role.Rules = []v1.PolicyRule{
+		{
+			APIGroups:     []string{"security.openshift.io"},
+			ResourceNames: []string{scc},
+			Resources:     []string{"securitycontextconstraints"},
+			Verbs:         []string{"use"},
+		},
+	}
+
+	_, err := oc.AdminKubeClient().RbacV1().Roles(role.ObjectMeta.Namespace).Update(ctx, role, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// creates or updates a namespace based on the options passed, then waits a bit for any controllers to apply
+// changes and returns an up-to-date namespace object
+func createOrUpdateNamespace(ctx context.Context, oc *exutil.CLI, ns *corev1.Namespace, options interface{}) (*corev1.Namespace, error) {
```

Comment:
You shouldn't need this function. Inline all of create/updates you have.

Reply:
fixed in 1ad30763012e28247539bc26eab52bef569de11c